pandas/io/common.py

```diff
@@ -153,6 +153,7 @@ def validate_header_arg(header) -> None:
 
 def stringify_path(
     filepath_or_buffer: FilePathOrBuffer[AnyStr],
+    convert_file_like: bool = False,
```

Comment:
umm, why would we *ever* want to stringify a Buffer?

Reply:
there is unfortunately one object (see the linked issue, I'm not sure whether there are more cases) that is `os.PathLike` and file-like. If the user provides such an object, we want to convert it to a string only to infer the compression but we want to use the existing object for reading/writing instead of opening the file ourselves.

Reply:
one option could be to remove this exception from `strinigfy_path` and simply check `__fspath__` in `infer_compression`? `infer_compression` is the only place that really needs to now the extension.